# House Price DataSet 

## Date:- 3 July 23

In [1]:
# Import the packages.
import numpy as np
import pandas as pd

In [2]:
# Load the dataset.
House_Price_Dataset = pd.read_csv("House Price India.csv")

In [3]:
# Task 1. Show the first 10 datasets.
House_Price_Dataset.head(10)
#head is used to display upward data.

id   Date  number of bedrooms  number of bathrooms  living area  \
0  6762810145  42491                   5                 2.50         3650   
1  6762810635  42491                   4                 2.50         2920   
2  6762810998  42491                   5                 2.75         2910   
3  6762812605  42491                   4                 2.50         3310   
4  6762812919  42491                   3                 2.00         2710   
5  6762813105  42491                   3                 2.50         2600   
6  6762813157  42491                   5                 3.25         3660   
7  6762813599  42491                   3                 1.75         2240   
8  6762813600  42491                   3                 2.50         2390   
9  6762814461  42491                   4                 2.25         2200   

   lot area  number of floors  waterfront present  number of views  \
0      9050               2.0                   0                4   
1      4000               1.5                   0                0   
2      9480               1.5                   0                0   
3     42998               2.0                   0                0   
4      4500               1.5                   0                0   
5      4750               1.0                   0                0   
6     11995               2.0                   0                2   
7     10578               2.0                   0                0   
8      6550               1.0                   0                2   
9     11250               1.5                   0                0   

   condition of the house  ...  Built Year  Renovation Year  Postal Code  \
0                       5  ...        1921                0       122003   
1                       5  ...        1909                0       122004   
2                       3  ...        1939                0       122004   
3                       3  ...        2001                0       122005   
4                       4  ...        1929                0       122006   
5                       4  ...        1951                0       122007   
6                       3  ...        2006                0       122008   
7                       5  ...        1923                0       122006   
8                       4  ...        1955                0       122009   
9                       5  ...        1920                0       122010   

   Lattitude  Longitude  living_area_renov  lot_area_renov  \
0    52.8645   -114.557               2880            5400   
1    52.8878   -114.470               2470            4000   
2    52.8852   -114.468               2940            6600   
3    52.9532   -114.321               3350           42847   
4    52.9047   -114.485               2060            4500   
5    52.9133   -114.590               2380            4750   
6    52.7637   -114.050               3320           11241   
7    52.9254   -114.482               1570           10578   
8    52.8014   -114.598               2010            6550   
9    52.9145   -114.391               2320           10814   

   Number of schools nearby  Distance from the airport    Price  
0                         2                         58  2380000  
1                         2                         51  1400000  
2                         1                         53  1200000  
3                         3                         76   838000  
4                         1                         51   805000  
5                         1                         67   790000  
6                         3                         72   785000  
7                         3                         71   750000  
8                         1                         73   750000  
9                         2                         53   698000  

[10 rows x 23 columns]

In [4]:
# Task 2 Show the last 10 Rows:-
House_Price_Dataset.tail(10)
## "tail" is a method used to display last data.

id   Date  number of bedrooms  number of bathrooms  \
14610  6762828349  42734                   4                 2.75   
14611  6762828783  42734                   3                 1.75   
14612  6762828856  42734                   3                 1.00   
14613  6762829600  42734                   3                 1.00   
14614  6762829669  42734                   3                 1.75   
14615  6762830250  42734                   2                 1.50   
14616  6762830339  42734                   3                 2.00   
14617  6762830618  42734                   2                 1.00   
14618  6762830709  42734                   4                 1.00   
14619  6762831463  42734                   3                 1.00   

       living area  lot area  number of floors  waterfront present  \
14610         1810      7350               1.0                   0   
14611         1350      7686               1.0                   0   
14612         1180      5350               1.5                   0   
14613         1400     10425               1.0                   0   
14614         1590      7931               1.0                   0   
14615         1556     20000               1.0                   0   
14616         1680      7000               1.5                   0   
14617         1070      6120               1.0                   0   
14618         1030      6621               1.0                   0   
14619          900      4770               1.0                   0   

       number of views  condition of the house  ...  Built Year  \
14610                0                       4  ...        1980   
14611                0                       3  ...        1987   
14612                0                       4  ...        1959   
14613                0                       4  ...        1968   
14614                0                       3  ...        1979   
14615                0                       4  ...        1957   
14616                0                       4  ...        1968   
14617                0                       3  ...        1962   
14618                0                       4  ...        1955   
14619                0                       3  ...        1969   

       Renovation Year  Postal Code  Lattitude  Longitude  living_area_renov  \
14610                0       122065    52.6003   -114.361               1750   
14611                0       122024    52.5917   -114.242               1370   
14612                0       122063    52.7350   -114.449               1490   
14613                0       122040    52.5038   -114.491               1440   
14614                0       122024    52.5928   -114.240               1680   
14615                0       122066    52.6191   -114.472               2250   
14616                0       122072    52.5075   -114.393               1540   
14617                0       122056    52.7289   -114.507               1130   
14618                0       122042    52.7157   -114.411               1420   
14619             2009       122018    52.5338   -114.552                900   

       lot_area_renov  Number of schools nearby  Distance from the airport  \
14610            7350                         1                         73   
14611            7686                         2                         70   
14612            5350                         3                         54   
14613           10425                         2                         59   
14614            7931                         1                         80   
14615           17286                         3                         76   
14616            7480                         3                         59   
14617            6120                         2                         64   
14618            6631                         3                         54   
14619            3480                         2                         55   

      

In [5]:
# Task 3 Show the Size if the Data.
House_Price_Dataset.shape

(14620, 23)

*Important*

`shape` is a method used to calculate size of the data.

**Inference of Task 3 **

- we have `14620` Observations (Rows).
- we have `23` Attributes (Columns).

# 5 July 23
# Steps to Get Inferance from the data or EDA(Exploratory Data Analysis):-

## Step 1:- Data Pre-Processing

## Step 2:- EDA

## Step 3 :- Draw the Conclusion

In [6]:
# Step 1: Data Pre-Processing:-

House_Price_Dataset.head(3)

id   Date  number of bedrooms  number of bathrooms  living area  \
0  6762810145  42491                   5                 2.50         3650   
1  6762810635  42491                   4                 2.50         2920   
2  6762810998  42491                   5                 2.75         2910   

   lot area  number of floors  waterfront present  number of views  \
0      9050               2.0                   0                4   
1      4000               1.5                   0                0   
2      9480               1.5                   0                0   

   condition of the house  ...  Built Year  Renovation Year  Postal Code  \
0                       5  ...        1921                0       122003   
1                       5  ...        1909                0       122004   
2                       3  ...        1939                0       122004   

   Lattitude  Longitude  living_area_renov  lot_area_renov  \
0    52.8645   -114.557               2880            5400   
1    52.8878   -114.470               2470            4000   
2    52.8852   -114.468               2940            6600   

   Number of schools nearby  Distance from the airport    Price  
0                         2                         58  2380000  
1                         2                         51  1400000  
2                         1                         53  1200000  

[3 rows x 23 columns]

In [7]:
## List out all column names 

House_Price_Dataset.columns

Index(['id', 'Date', 'number of bedrooms', 'number of bathrooms',
       'living area', 'lot area', 'number of floors', 'waterfront present',
       'number of views', 'condition of the house', 'grade of the house',
       'Area of the house(excluding basement)', 'Area of the basement',
       'Built Year', 'Renovation Year', 'Postal Code', 'Lattitude',
       'Longitude', 'living_area_renov', 'lot_area_renov',
       'Number of schools nearby', 'Distance from the airport', 'Price'],
      dtype='object')

In [8]:
## Check for  valid null values in dataset
## `isnull` is used to check valid null value (just blank) not invalid null value (jsut filled with anything) 
House_Price_Dataset.isnull().sum()

id                                       0
Date                                     0
number of bedrooms                       0
number of bathrooms                      0
living area                              0
lot area                                 0
number of floors                         0
waterfront present                       0
number of views                          0
condition of the house                   0
grade of the house                       0
Area of the house(excluding basement)    0
Area of the basement                     0
Built Year                               0
Renovation Year                          0
Postal Code                              0
Lattitude                                0
Longitude                                0
living_area_renov                        0
lot_area_renov                           0
Number of schools nearby                 0
Distance from the airport                0
Price                                    0
dtype: int6

** Inference** 
- We dont have any `valid null` values in the data.

## Access the Set Of Columns.

In [9]:
House_Price_Dataset.columns

Index(['id', 'Date', 'number of bedrooms', 'number of bathrooms',
       'living area', 'lot area', 'number of floors', 'waterfront present',
       'number of views', 'condition of the house', 'grade of the house',
       'Area of the house(excluding basement)', 'Area of the basement',
       'Built Year', 'Renovation Year', 'Postal Code', 'Lattitude',
       'Longitude', 'living_area_renov', 'lot_area_renov',
       'Number of schools nearby', 'Distance from the airport', 'Price'],
      dtype='object')

In [10]:
## Access the Home Coordinates
House_Price_Dataset_Home_Coordinates = House_Price_Dataset [['number of bedrooms', 'living area', 'number of bathrooms','condition of the house',  'Area of the house(excluding basement)', 'Area of the basement']]
House_Price_Dataset_Home_Coordinates.head(3)

number of bedrooms  living area  number of bathrooms  \
0                   5         3650                 2.50   
1                   4         2920                 2.50   
2                   5         2910                 2.75   

   condition of the house  Area of the house(excluding basement)  \
0                       5                                   3370   
1                       5                                   1910   
2                       3                                   2910   

   Area of the basement  
0                   280  
1                  1010  
2                     0

In [11]:
## Lattitude of house 52.8645
## Longitude of house -114.557



In [12]:
## Access the area coordinates.

House_Price_Dataset_Area_Coordinates = House_Price_Dataset[['waterfront present',
'Postal Code','Lattitude','Longitude', 'Number of schools nearby','Distance from the airport']]
House_Price_Dataset_Area_Coordinates.head(3)

waterfront present  Postal Code  Lattitude  Longitude  \
0                   0       122003    52.8645   -114.557   
1                   0       122004    52.8878   -114.470   
2                   0       122004    52.8852   -114.468   

   Number of schools nearby  Distance from the airport  
0                         2                         58  
1                         2                         51  
2                         1                         53

In [13]:
## Home Modification Coordinates.

House_Price_Dataset.columns

Index(['id', 'Date', 'number of bedrooms', 'number of bathrooms',
       'living area', 'lot area', 'number of floors', 'waterfront present',
       'number of views', 'condition of the house', 'grade of the house',
       'Area of the house(excluding basement)', 'Area of the basement',
       'Built Year', 'Renovation Year', 'Postal Code', 'Lattitude',
       'Longitude', 'living_area_renov', 'lot_area_renov',
       'Number of schools nearby', 'Distance from the airport', 'Price'],
      dtype='object')

In [14]:
House_Price_Dataset_Modification_Coordinates = House_Price_Dataset[['Renovation Year','living_area_renov','lot_area_renov']]
House_Price_Dataset_Modification_Coordinates.head(3)

Renovation Year  living_area_renov  lot_area_renov
0                0               2880            5400
1                0               2470            4000
2                0               2940            6600

In [15]:
## Establishment Coordinates.
House_Price_Dataset_Establishment_Coordinates = House_Price_Dataset[['id', 'Date','condition of the house', 'grade of the house','Built Year','number of views','Price']]
House_Price_Dataset_Establishment_Coordinates.head()

id   Date  condition of the house  grade of the house  Built Year  \
0  6762810145  42491                       5                  10        1921   
1  6762810635  42491                       5                   8        1909   
2  6762810998  42491                       3                   8        1939   
3  6762812605  42491                       3                   9        2001   
4  6762812919  42491                       4                   8        1929   

   number of views    Price  
0                4  2380000  
1                0  1400000  
2                0  1200000  
3                0   838000  
4                0   805000

## Univariate Analysis to authenticate the data :-
#  ` Univariate Analysis` shows how the data variates.

## A. Home Corodinates

In [16]:
House_Price_Dataset_Home_Coordinates.head(3)

number of bedrooms  living area  number of bathrooms  \
0                   5         3650                 2.50   
1                   4         2920                 2.50   
2                   5         2910                 2.75   

   condition of the house  Area of the house(excluding basement)  \
0                       5                                   3370   
1                       5                                   1910   
2                       3                                   2910   

   Area of the basement  
0                   280  
1                  1010  
2                     0

In [17]:
House_Price_Dataset_Home_Coordinates.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14620 entries, 0 to 14619
Data columns (total 6 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   number of bedrooms                     14620 non-null  int64  
 1   living area                            14620 non-null  int64  
 2   number of bathrooms                    14620 non-null  float64
 3   condition of the house                 14620 non-null  int64  
 4   Area of the house(excluding basement)  14620 non-null  int64  
 5   Area of the basement                   14620 non-null  int64  
dtypes: float64(1), int64(5)
memory usage: 685.4 KB


### `info()` shows the information about the data.

### ` describe()` is used to claculate the statistics of the data

 **Inference**
- In this home coordinate dataset we have all Columns Numerical in nature.
- In the same 2 columns as Float and reamining 6 are integer

In [18]:
House_Price_Dataset_Home_Coordinates.describe()

number of bedrooms   living area  number of bathrooms  \
count        14620.000000  14620.000000         14620.000000   
mean             3.379343   2098.262996             2.129583   
std              0.938719    928.275721             0.769934   
min              1.000000    370.000000             0.500000   
25%              3.000000   1440.000000             1.750000   
50%              3.000000   1930.000000             2.250000   
75%              4.000000   2570.000000             2.500000   
max             33.000000  13540.000000             8.000000   

       condition of the house  Area of the house(excluding basement)  \
count            14620.000000                           14620.000000   
mean                 3.430506                            1801.783926   
std                  0.664151                             833.809963   
min                  1.000000                             370.000000   
25%                  3.000000                            1200.000000   
50%                  3.000000                            1580.000000   
75%                  4.000000                            2240.000000   
max                  5.000000                            9410.000000   

       Area of the basement  
count          14620.000000  
mean             296.479070  
std              448.551409  
min                0.000000  
25%                0.000000  
50%                0.000000  
75%              580.000000  
max             4820.000000

In [19]:
House_Price_Dataset_Home_Coordinates.describe().T

count         mean         std  \
number of bedrooms                     14620.0     3.379343    0.938719   
living area                            14620.0  2098.262996  928.275721   
number of bathrooms                    14620.0     2.129583    0.769934   
condition of the house                 14620.0     3.430506    0.664151   
Area of the house(excluding basement)  14620.0  1801.783926  833.809963   
Area of the basement                   14620.0   296.479070  448.551409   

                                         min      25%      50%     75%  \
number of bedrooms                       1.0     3.00     3.00     4.0   
living area                            370.0  1440.00  1930.00  2570.0   
number of bathrooms                      0.5     1.75     2.25     2.5   
condition of the house                   1.0     3.00     3.00     4.0   
Area of the house(excluding basement)  370.0  1200.00  1580.00  2240.0   
Area of the basement                     0.0     0.00     0.00   580.0   

                                           max  
number of bedrooms                        33.0  
living area                            13540.0  
number of bathrooms                        8.0  
condition of the house                     5.0  
Area of the house(excluding basement)   9410.0  
Area of the basement                    4820.0

In [ ]:
## Now the time is to visualize the existing home coordinate  
# Import Seaborn library
import seaborn as sns


#Import Matplotlib library
import matplotlib.pyplot as plt


In [ ]:
#Draw the pairplot
sns.pairplot(House_Price_Dataset_Home_Coordinates)
plt.show()



### pairplot plots all the combinations of numeric columns from a dataset.
###  Here all attributes present in Home_Coordinates such as(no.of bedrroms,living area,no.of bathrooms etc) are plotted as
### combination by `pairplot.`

In [ ]:
#Draw the pairplot
sns.pairplot(House_Price_Dataset_Home_Coordinates,kind="hist")
plt.show()


**Interepretation**
- Living Area Vs Area Of House Excluding Basement.
- Area Of House Excluding Basement Vs no.of bathrooms.
- Living Area Vs No. Of Bathroom.
- No. of bedrooms Vs Living Area having gradual growth
- No.Of bathrooms Vs Condition of house/No.of FLoors (No Realtion).
- Living area vs Area of house having hetro scadasticity.
